x-pack/plugins/canvas/canvas_plugin_src/renderers/advanced_filter/component/advanced_filter.tsx

```diff
@@ -28,20 +34,20 @@ export const AdvancedFilter = ({ value, onChange, commit }) => (
         />
       </EuiFlexItem>
       <EuiFlexItem grow={false}>
-        <button
-          className="canvasAdvancedFilter__button"
-          type="submit"
-          onClick={() => commit(value)}
-        >
+        <button className="canvasAdvancedFilter__button" type="submit">
           Apply
         </button>
       </EuiFlexItem>
     </EuiFlexGroup>
   </form>
 );
 
+AdvancedFilter.defaultProps = {
+  value: '',
+};
+
 AdvancedFilter.propTypes = {
-  onChange: PropTypes.func,
+  onChange: PropTypes.func.isRequired,
   value: PropTypes.string,
```

Comment:
Completely unrelated to this PR, just trying to learn, and saw you've added component/type improvements: in this context, wouldn't it be useful to expect a `value`, even if it's the empty string? No specific reason, other than just simplifying types, `value` is likely coming from somewhere or could (should?) be initialized.